crates/weaver_forge/src/extensions/otel.rs

```diff
@@ -2,11 +2,40 @@
 
 //! Set of filters, tests, and functions that are specific to the OpenTelemetry project.
 
-use crate::config::CaseConvention;
 use itertools::Itertools;
 use minijinja::{ErrorKind, Value};
 use serde::de::Error;
 
+use crate::config::CaseConvention;
+
+/// Filters the input value to only include the required "object".
+/// A required object is one that has a field named "requirement_level" with the value "required".
+pub(crate) fn required(input: Value) -> Result<Vec<Value>, minijinja::Error> {
+    let mut rv = vec![];
+
+    for value in input.try_iter()? {
+        let required = value.get_attr("requirement_level")?;
+        if required.as_str() == Some("required") {
+            rv.push(value);
+        }
+    }
+    Ok(rv)
+}
+
+/// Filters the input value to only include the non-required "object".
+/// A optional object is one that has a field named "requirement_level" which is not "required".
+pub(crate) fn optional(input: Value) -> Result<Vec<Value>, minijinja::Error> {
```

Comment:
I'm not sure this is a great name as it conflicts with `opt-in`, and we've had confusion around that already.

Maybe just "not_required"

Reply:
Fixed